modules/nixos/general/keyboard.nix

```diff
@@ -0,0 +1,54 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.mine.keyboard;
+
+  layouts = {
+    azerty = {
+      layout = "fr";
+      variant = "";
+    };
+    bepo = {
+      layout = "fr";
+      variant = "bepo";
+    };
+    colemak = {
+      layout = "us";
+      variant = "colemak";
+    };
+    qwerty = {
+      layout = "us";
+      variant = "";
+    };
+    qwerty_intl = {
+      layout = "us";
+      variant = "intl";
+    };
+  };
+in
+
+with lib;
+
+{
+  options = {
+    mine.keyboard = {
+      layouts = mkOption {
+        type = types.listOf (types.enum (attrNames layouts));
+        default = [ "qwerty" ];
+        example = [ "colemak" "qwerty" ];
+        description = "Layouts to set";
+      };
+
+      enableAtBoot = mkEnableOption "If true, the first of the layouts will be enabled at boot";
+    };
+  };
+
+  config = {
+    boot.earlyVconsoleSetup = cfg.enableAtBoot;
+    i18n.consoleKeyMap = if layouts.(builtins.head cfg.layouts).variant == ""
+                         then layouts.(builtins.head cfg.layouts).layout
+                         else layouts.(builtins.head cfg.layouts).variant;
```

Comment:
Getting a syntax error on the CI [here](https://circleci.com/gh/kalbasit/shabka/2556?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link):

```
error: syntax error, unexpected '(', expecting ID or OR_KW or DOLLAR_CURLY or '"', at /root/shabka/modules/nixos/general/keyboard.nix:48:37
```

You probably want to create a local variable for `builtins.head cfg.layouts` in a let statement.

Reply:
Thanks for the review. I hadn't had time to check the results on the CI